test/unit/Fixture/ClassWithConstantsAsDefaultValues.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Roave\BetterReflectionTest\Fixture {
+
+	use Roave\BetterReflectionTest\FixtureOther\OtherClass;
+
+	class ClassWithConstantsAsDefaultValues
+	{
+		public const MY_CONST = 'my';
```

Comment:
Also note: a class constant could be defined multiple times at multiple levels of the inheritance. `self` always resolves to "this class", but a test is needed to check that behavior too.